### PR TITLE
feat: add ingress for azure

### PR DESCRIPTION
**Pull request by yasinlachiny**
# add ingress for azure

## Description

Create ingress for azure cloud

### Checklist

- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/particuleio/terraform-kubernetes-addons/#doc-generation

### CODE REVIEW

**Comment by ArchiFleKs** [comment]
Hi @yasinlachiny thanks and great to see a contribution for Azure :)

Do you mind recreating the symlink for shared component that are cloud provider agnostic. Like it is done [here](https://github.com/particuleio/terraform-kubernetes-addons/tree/main/modules/aws).

**Comment by yasinlachiny** [comment]
Hi @ArchiFleKs 
Thank you. The AWS part is so good that inspire me to contribute to the azure part.

Could you please explain it more? I checked the link that you shared with me.

symlink for shared components are  as below

```
-rw-r--r-- 1 root root 7273 Aug 31 12:40 README.md
lrwxrwxrwx 1 root root   28 Aug 31 12:40 helm-dependencies.yaml -> ../../helm-dependencies.yaml
-rw-r--r-- 1 root root 2499 Aug 31 12:40 ingress-nginx.tf
lrwxrwxrwx 1 root root   15 Aug 31 12:40 locals.tf -> ../../locals.tf
lrwxrwxrwx 1 root root   18 Aug 31 12:40 variables.tf -> ../../variables.tf
-rw-r--r-- 1 root root  378 Aug 31 12:40 version.tf
```

What parts should I change? which files?
Thank you in advance.

**Comment by ArchiFleKs** [comment]
@yasinlachiny you should relink in azure directory all the component that are present in the defaults `variables.tf` these are component present in all module, for example `metrics-server`. These component are in the root and then symlink to each provider modules

eg in the AWS folder:
```
lrwxrwxrwx     18 klefevre 30 Apr 15:42   admiralty.tf -> ../../admiralty.tf
.rw-rw-r--   9,2k klefevre 15 Aug 11:07   aws-ebs-csi-driver.tf
.rw-rw-r--   4,6k klefevre 15 Aug 11:16   aws-efs-csi-driver.tf
.rw-rw-r--   6,7k klefevre 10 Aug 00:56   aws-for-fluent-bit.tf
.rw-rw-r--   7,3k klefevre 10 Aug 00:56   aws-load-balancer-controller.tf
.rw-rw-r--   4,3k klefevre  6 Aug 12:42   aws-node-termination-handler.tf
.rw-rw-r--   3,4k klefevre 30 Apr 15:46   calico.tf
.rw-rw-r--  10,0k klefevre 10 Aug 00:56   cert-manager.tf
.rw-rw-r--   8,0k klefevre 10 Aug 00:56   cluster-autoscaler.tf
.rw-rw-r--   2,1k klefevre 15 Aug 10:22   cni-metrics-helper.tf
.rw-rw-r--     72 klefevre 30 Apr 15:46   data.tf
drwxrwxr-x      - klefevre 14 Aug 14:58   examples/
.rw-rw-r--   6,6k klefevre 14 Aug 15:40   external-dns.tf
lrwxrwxrwx     13 klefevre 30 Apr 15:42   flux.tf -> ../../flux.tf
lrwxrwxrwx     14 klefevre 30 Apr 15:42   flux2.tf -> ../../flux2.tf
lrwxrwxrwx     28 klefevre 30 Apr 15:42   helm-dependencies.yaml -> ../../helm-dependencies.yaml
drwxrwxr-x      - klefevre 15 Aug 11:16   iam/
.rw-rw-r--   9,7k klefevre 14 Aug 16:10   ingress-nginx.tf
lrwxrwxrwx     23 klefevre  5 Jan 22:09   istio-operator.tf -> ../../istio-operator.tf
lrwxrwxrwx     13 klefevre 12 Mar 10:41   k8gb.tf -> ../../k8gb.tf
lrwxrwxrwx     14 klefevre  5 Jan 22:09   karma.tf -> ../../karma.tf
lrwxrwxrwx     13 klefevre 21 Jul 10:26   keda.tf -> ../../keda.tf
lrwxrwxrwx     17 klefevre  5 Jan 22:09   keycloak.tf -> ../../keycloak.tf
lrwxrwxrwx     18 klefevre 12 Jul 09:21   kong-crds.tf -> ../../kong-crds.tf
lrwxrwxrwx     13 klefevre  8 Mar 12:32   kong.tf -> ../../kong.tf
lrwxrwxrwx     28 klefevre 12 Jul 09:21   kube-prometheus-crd.tf -> ../../kube-prometheus-crd.tf
.rw-rw-r--    19k klefevre 15 Aug 10:28   kube-prometheus.tf
lrwxrwxrwx     16 klefevre  4 Mar 08:41   kyverno.tf -> ../../kyverno.tf
.rw-rw-r--     29 klefevre 30 Apr 15:46   locals-aws.tf
lrwxrwxrwx     15 klefevre  5 Jan 22:09   locals.tf -> ../../locals.tf
.rw-rw-r--    10k klefevre  1 Sep 08:57   loki-stack.tf
lrwxrwxrwx     23 klefevre  5 Jan 22:09   metrics-server.tf -> ../../metrics-server.tf
lrwxrwxrwx     30 klefevre  5 Jan 22:09   node-problem-detector.tf -> ../../node-problem-detector.tf
lrwxrwxrwx     23 klefevre  5 Jan 22:09   priority-class.tf -> ../../priority-class.tf
lrwxrwxrwx     27 klefevre  5 Mar 18:13   prometheus-adapter.tf -> ../../prometheus-adapter.tf
lrwxrwxrwx     37 klefevre  5 Mar 18:13   prometheus-blackbox-exporter.tf -> ../../prometheus-blackbox-exporter.tf
.rw-rw-r--   7,3k klefevre 10 Aug 00:56   prometheus-cloudwatch-exporter.tf
lrwxrwxrwx     17 klefevre 30 Aug 16:07   promtail.tf -> ../../promtail.tf
.rw-rw-r--    51k klefevre 30 Aug 16:44   README.md
lrwxrwxrwx     23 klefevre  5 Jan 22:09   sealed-secrets.tf -> ../../sealed-secrets.tf
.rw-rw-r--   1,3k klefevre 26 Aug 06:05   secrets-store-csi-driver-provider-aws.tf
lrwxrwxrwx     33 klefevre 26 Aug 05:37   secrets-store-csi-driver.tf -> ../../secrets-store-csi-driver.tf
lrwxrwxrwx     31 klefevre 18 Feb 08:37   strimzi-kafka-operator.tf -> ../../strimzi-kafka-operator.tf
drwxrwxr-x      - klefevre 15 Aug 10:21   templates/
.rw-rw-r--   2,4k klefevre 30 Apr 15:46   thanos-memcached.tf
.rw-rw-r--   5,4k klefevre 10 Aug 00:56   thanos-storegateway.tf
.rw-rw-r--   4,8k klefevre 30 Apr 15:46   thanos-tls-querier.tf
.rw-rw-r--    13k klefevre 10 Aug 00:56   thanos.tf
.rw-rw-r--   4,1k klefevre 14 Aug 14:58   tigera-operator.tf
.rw-rw-r--   1,8k klefevre 21 Jul 10:26   variables-aws.tf
lrwxrwxrwx     18 klefevre  5 Jan 22:09   variables.tf -> ../../variables.tf
.rw-rw-r--   9,1k klefevre 26 Aug 05:26   vault.tf
.rw-rw-r--    393 klefevre 31 Aug 14:21   versions.tf
```

All the symlinked component here are component that can work anywhere to to avoid duplication and make them available for azure you can symlink them. If not, I can do it after merging this, maybe it will be clearer.

**Comment by yasinlachiny** [comment]
Hi @ArchiFleKs
Thanks, your explanation clear the issue for me.
I added all of them except `metrics-server` because [metrics-server ](https://github.com/Azure/aks-engine/blob/master/docs/topics/monitoring.md)is part of AKS Engine deployment.

File: modules/azure/version.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "2.66.0"
```
Comment (ArchiFleKs) [inline modules/azure/version.tf:6]: Could you use a conditionnal like the other ? ether `~`  or ` >=`

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

**Comment by ArchiFleKs** [comment]
And also rerun the pre-commit with latest rebase

File: modules/azure/version.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    azurerm    = "~> 2.66.0"
```
Comment (ArchiFleKs) [inline modules/azure/version.tf:4]: This is going to lock to 2.66.X, could you put `~> 2.0`

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

File: modules/azure/version.tf (hunk 2 above)
Comment (yasinlachiny) [reply modules/azure/version.tf:4]: Yes, of course. I fix it now.

**Review by ArchiFleKs — APPROVED** [review]

**Pull request merged**